src/atoms/affine/partialtrace.jl

```diff
@@ -0,0 +1,85 @@
+import Base.sign
+export partialtrace
+
+type PartialTraceAtom <: AbstractExpr
+	head::Symbol
+	id_hash::UInt64
+	children::Tuple{AbstractExpr}
+	size::Tuple{Int, Int}
+	sys::Int
+	dims::Vector
+
+	function PartialTraceAtom(x::AbstractExpr, sys::Int, dims::Vector)
+		if x.size[1] ≠ x.size[2]
+			error("Only square matrices are supported")
+		end
+
+		if ! (1 ≤ sys ≤ length(dims))
+			error("Invalid system, should between 1 and ", length(dims), " got ", sys)
+		end
+
+		if x.size[1] ≠ prod(dims)
```

Comment:
Since x is the tensor product of the two matrices, this condition checks whether the dims of the subsystems is correct.